addons/point_of_sale/models/product_pricelist.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class PosPayment(models.Model):
+    _inherit = "product.pricelist"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'active' in vals and not vals['active'] and self.env['pos.config'].sudo().with_context(active_test=False).search_count([('pricelist_id', 'in', self.ids)]):
```

Comment:
do your stuff before the call of super, as it'll avoid to write and rollback it afterward